packages/core/src/animation/Animator.ts

```diff
@@ -872,6 +881,30 @@ export class Animator extends Component {
     }
   }
 
+  private _fireBackwardSubAnimationEvents(
+    playState: AnimatorStatePlayData,
+    eventHandlers: AnimationEventHandler[],
+    lastClipTime: number,
+    curClipTime: number
+  ): void {
+    for (let i = playState.currentEventIndex, n = eventHandlers.length; i < n; i++) {
+      const eventHandler = eventHandlers[n - 1 - i];
+      const { time, parameter } = eventHandler.event;
+
+      if (time < curClipTime) {
+        break;
+      }
+
+      const { handlers } = eventHandler;
+      if (time <= lastClipTime) {
+        for (let j = handlers.length - 1; j >= 0; j--) {
+          handlers[j](parameter);
+        }
+        playState.currentEventIndex = i + 1;
+      }
+    }
+  }
+
```

Comment:
- reduce assignment of `playState.currentEventIndex`;
- Use more nature loop logic to optimization `_fireBackwardSubAnimationEvents`

```ts
private _fireSubAnimationEvents(
    playState: AnimatorStatePlayData,
    eventHandlers: AnimationEventHandler[],
    lastClipTime: number,
    curClipTime: number
  ): void {
    let eventIndex = playState.currentEventIndex;
    for (let n = eventHandlers.length; eventIndex < n; eventIndex++) {
      const eventHandler = eventHandlers[eventIndex];
      const { time, parameter } = eventHandler.event;

      if (time > curClipTime) {
        break;
      }

      const { handlers } = eventHandler;
      if (time >= lastClipTime) {
        for (let j = handlers.length - 1; j >= 0; j--) {
          handlers[j](parameter);
        }
      }
    }
    playState.currentEventIndex = eventIndex;
  }

  private _fireBackwardSubAnimationEvents(
    playState: AnimatorStatePlayData,
    eventHandlers: AnimationEventHandler[],
    lastClipTime: number,
    curClipTime: number
  ): void {
    let eventIndex = playState.currentEventIndex;
    for (; eventIndex >= 0; eventIndex--) {
      const eventHandler = eventHandlers[eventIndex];
      const { time, parameter } = eventHandler.event;

      if (time < curClipTime) {
        break;
      }

      const { handlers } = eventHandler;
      if (time <= lastClipTime) {
        for (let j = handlers.length - 1; j >= 0; j--) {
          handlers[j](parameter);
        }
      }
    }
    playState.currentEventIndex = eventIndex;
  }
```

Reply:
> ```ts
>     playState.currentEventIndex = eventIndex;
> ```
When trigger break, i will be wrong.